src/shadowbox/server/manager_service.ts

```diff
@@ -156,9 +161,9 @@ export class ShadowsocksManagerService {
       Promise<void> {
     try {
       logging.debug(`setPortForNewAccessKeys request ${JSON.stringify(req.params)}`);
-      const port = req.params.port;
-      if (typeof port !== 'number') {
-        return next(new restify.InvalidArgumentError(port));
+      const port = Number.parseFloat(req.params.portStr);
```

Comment:
Use `Number.parseInt`

Reply:
parseInt doesn't work, as it parses decimals as rounded to the nearest integer when we want those inputs to be rejected.